feat: support `RETURNING` clause

ginnwork opened the pull request:
Resolves #138

pi0 commented:
Can you please add a test? 🙏🏼

nick-w-nick commented:
@ginnwork the `RETURNING` clause is specific to Postgres, can you make this non-generic so it only applies to the applicable dialect(s) and can be overridden as needed?

ginnwork commented:
I updated the pull request with the changes @nick-w-nick suggested and added the tests @pi0 requested.

Was there any other changes needed to merge this pull request?

pi0 reviewed (approved):
Thanks ❤️

ginnwork commented:
Glad I could help!
